prusti-specs/src/lib.rs

```diff
@@ -435,3 +450,75 @@ pub fn extern_spec(_attr: TokenStream, tokens:TokenStream) -> TokenStream {
         _ => { unimplemented!() }
     }
 }
+
+pub fn predicate(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+    if !attr.is_empty() {
+        return syn::Error::new(
+            attr.span(),
+            "the `#[predicate]` attribute does not take parameters"
+        ).to_compile_error();
+    }
+
+    // emit a custom error to the user instead of a parse error
+    let mut item: untyped::AnyFnItem = handle_result!(
+        syn::parse2(tokens)
+            .map_err(|e| syn::Error::new(
+                e.span(),
+                "you can only use `#[predicate]` on function definitions"
+            ))
+    );
+    let item_span = item.span();
+
+    // check for other attributes -- #[predicate] is incompatible with all others currently
+    // make sure to also update the check in `rewrite_prusti_attributes`
+    let other_prusti_attrs = extract_prusti_attributes(&mut item);
+    if other_prusti_attrs.count() != 0 {
```

Comment:
```suggestion
    if !other_prusti_attrs.is_empty() {
```

Reply:
`extract_prusti_attributes` returns an `impl Iter`, not a `Vec`, which doesn't seem to have an `is_empty` method.